### PR TITLE
MVN nspc kernel optimization

### PR DESCRIPTION
**Pull request by chenhu-wang**
### Details:
 - *MVN nspc kernel optimization*
 - *fast path for mean calculation to I8/U8/BF16 precision with vnni and dp_bf16 for continuous data*

### Tickets:
 - *88026*

### CODE REVIEW

**Comment by chenhu-wang** [comment]
@dmitry-gorokhov , could please take a look? Previously nCsp8[16]c and nspc kernel have similar implementation to reuse basic code and schedule. It's not optimal for nspc kernel. This optimization provide a more memory friendly method.

**Comment by dmitry-gorokhov** [comment]
@xuchen May I ask you review? Thanks

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/mvn.cpp [hunk 1]
```diff
@@ -433,13 +433,15 @@ const std::vector<ov::Shape> inputShapesStatic_3D = {
 const std::vector<ov::Shape> inputShapesStatic_4D = {
         {1, 7, 3, 5},
         {1, 15, 9, 5},
-        {4, 41, 6, 9}
+        {4, 41, 6, 9},
+        {1, 4 * 16 * 2 + 16 + 3, 2, 2}
```
Comment (xuchen-intel) [inline src/plugins/intel_cpu/tests/functional/single_layer_tests/mvn.cpp:437]: Maybe we can just use a number directly. Decriptions can be added if necessary.

File: src/plugins/intel_cpu/src/nodes/mvn.cpp [hunk 2]
```diff
@@ -1261,6 +1667,172 @@ void MVN::MVNRefExecutor::mvn_ref(const uint8_t* src_data, uint8_t* dst_data) {
...
+                    (*mvn_kernel)(&arg);
+                });
+            }
+        } else {  // for per_channel
+            float size_inv = 1.f / static_cast<float>(D * H * W);
+            parallel_nt(0, [&](const int ithr, const int nthr) {
+                size_t start = 0, end = 0;
+                splitter(D * H * W, nthr, ithr, start, end);
+                auto mean_buffer_ptr = &mean_buffer[aux_buffer_size * ithr];
+                memset(mean_buffer_ptr, 0, aux_buffer_size * sizeof(float));
```
Comment (xuchen-intel) [inline src/plugins/intel_cpu/src/nodes/mvn.cpp:1761]: Does it make sense to apply such memset variance_buffer in all the other similar if..else... branches?

File: src/plugins/intel_cpu/src/nodes/mvn.cpp [hunk 3]
```diff
@@ -1261,6 +1667,172 @@ void MVN::MVNRefExecutor::mvn_ref(const uint8_t* src_data, uint8_t* dst_data) {
...
+                    arg.dst = dst_data + (b_offset + (start * C)) * dst_data_size;
+                    arg.mean = &mean_buffer[0];
+                    arg.variance = &variance_buffer[0];
+                    arg.work_amount = static_cast<size_t>(end - start);
+                    arg.oc_off = 0;
+                    arg.post_op_data = post_ops_data_;
+                    (*mvn_kernel)(&arg);
+                });
+            } else {
+                parallel_nt(0, [&](const int ithr, const int nthr) {
```
Comment (xuchen-intel) [inline src/plugins/intel_cpu/src/nodes/mvn.cpp:1741]: This code block seems to be duplicated in several places. Can we make it a function to be reused.

**Review by xuchen-intel — CHANGES_REQUESTED** [review]

File: src/plugins/intel_cpu/src/nodes/mvn.cpp [hunk 4]
```diff
@@ -1261,6 +1667,172 @@ void MVN::MVNRefExecutor::mvn_ref(const uint8_t* src_data, uint8_t* dst_data) {
...
+    size_t aux_buffer_size = mvnAttrs.execAcrossChannels_ ? 1 : rnd_up(C, blk_size);
+    parallel_for(N, [&](size_t b) {
+        std::vector<float> mean_buffer(aux_buffer_size * threads_num);
+        std::vector<float> variance_buffer(aux_buffer_size * threads_num);
+        size_t b_offset = b * C * D * H * W;
+        if (mvnAttrs.execAcrossChannels_) {
+            float size_inv = 1.f / static_cast<float>(C * D * H * W);
+            parallel_nt(0, [&](const int ithr, const int nthr) {
+                size_t start = 0, end = 0;
+                splitter(D * H * W, nthr, ithr, start, end);
```
[the remark this answers is not in the document]
Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/nodes/mvn.cpp:1697]: Both cases only have at most one tails for each thread. For mean and variance, the work_amount is (end-start * C), proceed as continue data. Normalize is channel sensitive(post-ops), better no split to different kernels.

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/mvn.cpp (hunk 1 above)
Comment (chenhu-wang) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/mvn.cpp]: comment is used. thanks

File: src/plugins/intel_cpu/src/nodes/mvn.cpp (hunk 2 above)
Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/nodes/mvn.cpp]: memset is not need any more for all case for current kernel, removed.

File: src/plugins/intel_cpu/src/nodes/mvn.cpp (hunk 3 above)
Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/nodes/mvn.cpp]: extracted, thanks.

**Review by xuchen-intel — APPROVED** [review]

**Pull request merged**